src/components/box/Box.tsx

```diff
@@ -1,78 +1,129 @@
+  padding: typeof paddings[number];
+  paddingHorizontal: typeof paddings[number];
+  paddingVertical: typeof paddings[number];
+  paddingBottom: typeof paddings[number];
+  paddingLeft: typeof paddings[number];
+  paddingRight: typeof paddings[number];
+  paddingTop: typeof paddings[number];
+  style: CSSProperties;
+  textAlign: 'center' | 'left' | 'right';
+}>;
 
-  const {
-    alignContent,
-    alignItems,
-    alignSelf,
-    backgroundColor,
-    backgroundTint,
-    borderBottomWidth,
-    borderColor,
-    borderLeftWidth,
-    borderRightWidth,
-    borderTint,
-    borderTopWidth,
-    borderWidth,
-    borderRadius,
-    borderTopLeftRadius = borderRadius,
-    borderTopRightRadius = borderRadius,
-    borderBottomLeftRadius = borderRadius,
-    borderBottomRightRadius = borderRadius,
-    boxSizing,
-    children,
-    className,
-    display,
-    element,
-    flex,
-    flexBasis,
-    flexDirection,
-    flexGrow,
-    flexShrink,
-    flexWrap,
-    forwardedRef,
-    justifyContent,
-    margin,
-    marginHorizontal = margin,
-    marginVertical = margin,
-    marginBottom = marginVertical,
-    marginLeft = marginHorizontal,
-    marginRight = marginHorizontal,
-    marginTop = marginVertical,
-    order,
-    overflow,
-    overflowX,
-    overflowY,
-    padding,
-    paddingHorizontal = padding,
-    paddingVertical = padding,
-    paddingBottom = paddingVertical,
-    paddingLeft = paddingHorizontal,
-    paddingRight = paddingHorizontal,
-    paddingTop = paddingVertical,
-    style,
-    textAlign,
-    ...others
-  } = props;
+const Box = ({
+  alignContent,
+  alignItems,
+  alignSelf,
+  backgroundColor,
+  backgroundTint,
+  borderBottomWidth,
+  borderColor = 'neutral',
+  borderLeftWidth,
+  borderRightWidth,
+  borderTint = 'dark',
+  borderTopWidth,
+  borderWidth,
+  borderRadius = 'square',
+  borderTopLeftRadius = borderRadius,
+  borderTopRightRadius = borderRadius,
+  borderBottomLeftRadius = borderRadius,
+  borderBottomRightRadius = borderRadius,
+  boxSizing,
+  children,
+  className,
+  display,
+  element = 'div',
+  flex,
+  flexBasis,
+  flexDirection,
+  flexGrow,
+  flexShrink,
+  flexWrap,
+  forwardedRef,
+  justifyContent,
+  margin = 0,
+  marginHorizontal = margin,
+  marginVertical = margin,
+  marginBottom = marginVertical,
+  marginLeft = marginHorizontal,
+  marginRight = marginHorizontal,
+  marginTop = marginVertical,
+  order,
+  overflow,
+  overflowX,
+  overflowY,
+  padding = 0,
+  paddingHorizontal = padding,
+  paddingVertical = padding,
+  paddingBottom = paddingVertical,
+  paddingLeft = paddingHorizontal,
+  paddingRight = paddingHorizontal,
+  paddingTop = paddingVertical,
+  style,
+  textAlign,
+  ...others
+}: Props) => {
+  const getBorder = (value: number) => {
+    return `${value}px solid ${
+      COLOR[borderColor.toUpperCase() as Uppercase<keyof typeof COLOR>][
+        borderTint.toUpperCase() as Uppercase<keyof typeof COLOR['AQUA']>
```

Comment:
PP define the colors using enums instead of having to try really hard to make the types work

example: https://www.typescriptlang.org/play?#code/KYOwrgtgBAKgliALgZygbwFBSgGTgcwAtFhlEoBeKAchwEkBxACRgFEBlGagGi1wOKUa9Zl17YAcgHsAThACGAGyHUJAeQBKAWQCCOHnwAi8mQGsVhnRoDSB7MbOlyVapZscxGAL4YMoSFAAwlKKsqiY2DoAjmDyKjoAigCqOnZQDCEAJioMajiGaRLAYIgySioSrEkwGnppWgjONFp0Ep7YGmAARgCeKhpJAEIAmmkwwOUubHXiUABqcCHATdRzdHmsnj4YAMZSIGRBeZoAXFAawHsymQA8waEyyNznl7K38EhPUGQyCPgAfP8hBEoABte5hAB00ViAF0ziDsKCPihIXgiCQyPCaABiABmACY8QAOPF4tJIlHINECRDY6g4nY7YAAFjJFLBVMh0jkSnpOIAnAKdgL2bNKY1qQ5TPyAAyygDsAEZCRzkZLIdKnHLZSzMvIAGxpLyzAD0pqghmAeIQwCgUkQhGAMigeweyG8AG5fHsDuR8MtBm9nUIABQAdzgmUdZ3AEC6zuebtkZwhj2eiEaZypAEpKECQTJlmAZCAoAADAAkaEj0cIXgADgAPb4hKNQauBY4aUHJmSw0GZpCwrzl7w+-bIJaQ0L4UMBxBB67O0NK55p6kw+TPLk8hSKHM5z1AA

Reply:
On second thought the lowercase / uppercase will still be required to prevent changing the api. It's still slightly nicer than the `keyof typeof` thing IMO

```
const getBorder = (width: number, color: Lowercase<Colors> | Colors, tint: Lowercase<Tints> | Tints) => {
  return `${width}px solid ${COLOR[color.toUpperCase() as Colors][tint.toUpperCase() as Tints]}`
}

// New ts way
console.log(getBorder(1, Colors.Aqua, Tints.Normal));

// Old js way
console.log(getBorder(1, 'aqua', 'normal'));
```

Reply:
Good suggestion, was planning on providing good generics in a separate PR, created something in the backlog for it